Framework/DataHandling/test/LoadNexusLogsTest.h

```diff
@@ -292,6 +293,60 @@ class LoadNexusLogsTest : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS(endTime.totalNanoseconds(), lastTime.totalNanoseconds());
   }
 
+  void test_load_file_with_invalid_log_entries() {
+    LoadNexusLogs ld;
+    ld.initialize();
+    ld.setPropertyValue("Filename", "ENGINX00228061_log_alarm_data.nxs");
+    MatrixWorkspace_sptr ws = createTestWorkspace();
+    // Put it in the object.
+    ld.setProperty("Workspace", ws);
+    ld.execute();
+    TS_ASSERT(ld.isExecuted());
+
+    auto run = ws->run();
+
+    // This one should not be present as there is no invalid data
+    TS_ASSERT_THROWS_ANYTHING(
+        run.getLogData(LogManager::getInvalidValuesFilterLogName("slitpos")));
+
+    // This one should not be present as there is no invalid data
+    TS_ASSERT_THROWS_ANYTHING(run.getLogData(
+        LogManager::getInvalidValuesFilterLogName("cryo_Sample")));
+
+    // these two both contain invalid data
+    auto pclog1 = dynamic_cast<TimeSeriesProperty<bool> *>(run.getLogData(
+        LogManager::getInvalidValuesFilterLogName("cryo_temp1")));
+    std::vector<bool> correct{true, false, true};
+    TS_ASSERT_EQUALS(pclog1->valuesAsVector(), correct);
+
+    auto pclog2 = dynamic_cast<TimeSeriesProperty<bool> *>(run.getLogData(
+        LogManager::getInvalidValuesFilterLogName("cryo_temp2")));
+    std::vector<bool> correct2{false, false, false};
+    TS_ASSERT_EQUALS(pclog2->valuesAsVector(), correct2);
```

Comment:
It looks to me as if this is the line causing the issue for `AppleClang`:
```
DataHandling/test/LoadNexusLogsTest.h:320:5: note: in instantiation of function template specialization 'CxxTest::doAssertEquals<std::__1::vector<bool, std::__1::allocator<bool> >, std::__1::vector<bool, std::__1::allocator<bool> > >' requested here
TS_ASSERT_EQUALS(pclog1->valuesAsVector(), correct);
```
`vector<bool>` is specialized type as it tries to be memory efficient by packing 8 bits into a word but it causes issues with things assuming you can just access things directly in the array as you might naively expect. For some reason AppleClang will need special handling inside CxxTest to be able to print it. An easier work around for this would be

```c++
const auto pcValues = pclog2->valuesAsVector()
TS_ASSERT_EQUALS(pcValues.size(), correct2.size());
for(int i = 0; i < pcValues.size; ++i) {
  TS_ASSERT_EQUALS(pcValues[i], correct2[i]);
}
```
for now.